src/Modules/TeamInfoModule/Controllers/TeamInfoEventController.cs

```diff
@@ -78,4 +92,15 @@ public async Task OnEndMapAsync(object sender, MapGbxEventArgs args)
 
         await teamInfoService.HideTeamInfoWidgetEveryoneAsync();
     }
+
+    [Subscribe(TeamSettingsEvents.SettingsUpdated)]
+    public async Task OnTeamSettingsUpdatedAsync(object sender, TeamSettingsEventArgs args)
+    {
+        if (!await teamInfoService.GetModeIsTeamsAsync())
+        {
+            return;
+        }
+        
+        await Task.Delay(250).ContinueWith(t => teamInfoService.SendTeamInfoWidgetEveryoneAsync());
```

Comment:
What is this delay for?

Reply:
Team info uses the club link retrieved through RPC and it can take around 100ms before a new club link gets processed by the dedicated.
I found that sometimes the info won't update without the delay.

Reply:
Okay, then i suggest you make this event handler async so that it doesnt block for 250ms.